### ISSUE TITLE
`runMigrationSilent` is not threadsafe

### ISSUE THREAD

**Issue by NorfairKing** [yesodweb/persistent#966]
Example errors:
```       
/dev/null: hDuplicateTo: resource busy (Device or resource busy)
```
```
<stderr>: hDuplicateTo: resource busy (Device or resource busy)
```

This is because of the call to `hSilence [stderr]` which is not threadsafe.
This call was added in response to #105  because of the way the postgres backend works.
I'm using `sqlite`, and with that call removed in #965, everything "Just works":tm: but that doesn't solve the problem for postgres.
#105 mentions that you can silence postgres by setting `PGOPTIONS` to `--client-min-messages=warning`.

So a few options:

- Leave `runMigrationSilent` non-threadsafe. Then I'll use `runMigration` and just get a bunch of nonsense output: 
![welsh](https://user-images.githubusercontent.com/3521180/66295719-46d80980-e8ec-11e9-99a8-cd655fc8b375.png)
- Split `runMigrationSilent` into a version that uses `hSilence` and one that doesn't.
- Remove the call to `hSilence` and instruct postgres users to set `PGOPTIONS` accordingly.

I'm in favour of the second, with some good docs.
I could do that as part of #965

**Comment by parsonsmatt** [comment]
Fixed by #971

**Comment by NorfairKing** [comment]
Wonderful!
